commercetools-models/src/main/java/io/sphere/sdk/carts/Cart.java

```diff
@@ -187,6 +187,16 @@ public String toString() {
     List<Reference<CartDiscount>> getRefusedGifts();
 
     /**
+     *  The shippingRateInput is used as an input to select a shipping rate price tier
+     * @return shippingRateInput
+     */
+    @Override
+    @Nullable
+    @HasUpdateAction
+    @QueryModelHint(type = "ShippingRateInputQueryModel<Cart>")
```

Comment:
I never used the `@QueryModelHint` annotation, may be you can explain it to me? 😉

Reply:
apparently when you generate a QueryModel, there are a number of rules checked by the generator for the target class, among this rules `UseQueryModelHintRule` wich checkos if a queryable field has `@QueryModelHint` annotation

Reply:
Thanks for your explanation 👍